### PR TITLE
Feature/ted 68 - almost ready notice domain model

### PR DESCRIPTION
**Pull request by costezki**
- unit tests complete
- functional tests partially complete but commitable
- the adoption of TOX for testing
- fixed notice status transitions
- polished the Makefile
- includes pulls from several (parallel) branches

### CODE REVIEW

File: requirements.dev.txt [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
-tox-envreport
-tox-pytest-summary
 coverage
 pytest
 pytest-bdd
 pytest-cov
 pytest-subtests
-
+tox
+tox-pytest-summary
```
Comment (CaptainOfHacks) [inline requirements.dev.txt:8]: freeze versions for libraries

File: ted_sws/domain/model/notice.py [hunk 2]
```diff
@@ -42,21 +49,41 @@ class NoticeStatus(Enum):
...
-                    NoticeStatus.PACKAGED: [NoticeStatus.FAULTY_PACKAGE, NoticeStatus.CORRECT_PACKAGE],
-                    NoticeStatus.CORRECT_PACKAGE: [NoticeStatus.PUBLISHED],
-                    NoticeStatus.PUBLISHED: [NoticeStatus.PUBLICLY_AVAILABLE, NoticeStatus.PUBLICLY_UNAVAILABLE]}
-
-
-class WorkExpression(PropertyBaseModel):
+    def __lt__(self, other):
+        if type(other) == type(self):
+            return self.value < other.value
+        raise ValueError(f"Cannot compare {self.name} and {other.name}")
```
Comment (CaptainOfHacks) [inline ted_sws/domain/model/notice.py:55]: add tests for coverage exceptions

File: ted_sws/domain/model/notice.py [hunk 3]
```diff
@@ -42,21 +49,41 @@ class NoticeStatus(Enum):
...
+
+    def __gt__(self, other):
+        if type(other) == type(self):
+            return self.value > other.value
+        raise ValueError(f"Cannot compare {self.name} and {other.name}")
```
Comment (CaptainOfHacks) [inline ted_sws/domain/model/notice.py:60]: add tests for coverage exceptions

File: ted_sws/domain/model/notice.py [hunk 4]
```diff
@@ -78,26 +105,22 @@ class Config:
...
-        :param value:
+        :param new_status:
         :return:
         """
-
-        if value in NOTICE_STATE_MAP[self._status]:
-            self._status = value
-        else:
-            raise Exception(f"Invalid transition from state {self._status} to state {value}")
+        pass
```
Comment (CaptainOfHacks) [inline ted_sws/domain/model/notice.py:123]: remove "pass" from ABC method

File: ted_sws/domain/model/notice.py [hunk 5]
```diff
@@ -138,35 +163,69 @@ def rdf_manifestation(self):
...
     def set_rdf_manifestation(self, rdf_manifestation: RDFManifestation):
         """
-
+            Add an RDF manifestation to the notice.
+            If METS package data are available, erase them and reset the state.
         :param rdf_manifestation:
         :return:
         """
+        if self.rdf_manifestation is rdf_manifestation:
+            return
```
Comment (CaptainOfHacks) [inline ted_sws/domain/model/notice.py:187]: add tests for coverage exceptions

**Review by CaptainOfHacks — APPROVED** [review]

File: requirements.dev.txt [hunk 6]
```diff
@@ -1,8 +1,8 @@
...
-tox-pytest-summary
\ No newline at end of file
+# Development libraries
+coverage~=6.3.1
+pytest~=7.0.0
+pytest-bdd~=5.0.0
+pytest-cov~=3.0.0
+pytest-subtests~=0.6.0
+tox~=3.24.5
+tox-pytest-summary~=0.1.2
```
Comment (CaptainOfHacks) [inline requirements.dev.txt:8]: Well done! 👍

**Review by kaleanych — APPROVED** [review]
I trust the experts! :D

**Pull request merged**